packages/date-picker/src/DatePicker/DatePicker.spec.tsx

```diff
@@ -102,10 +102,34 @@ describe('packages/date-picker', () => {
         expect(menuContainerEl).not.toBeInTheDocument();
       });
 
+      test('menu is initially closed when rendered with `initialOpen` and `disabled`', async () => {
+        const { findMenuElements } = renderDatePicker({
+          initialOpen: true,
+          disabled: true,
+        });
+        const { menuContainerEl } = await findMenuElements();
+        expect(menuContainerEl).not.toBeInTheDocument();
+      });
+
       test('menu is initially open when rendered with `initialOpen`', async () => {
         const { findMenuElements } = renderDatePicker({ initialOpen: true });
         const { menuContainerEl } = await findMenuElements();
-        await waitFor(() => expect(menuContainerEl).toBeInTheDocument());
+        expect(menuContainerEl).toBeInTheDocument();
+      });
+
+      test('open menu closes when `disabled` is set to true and fires handler', async () => {
```

Comment:
I would break this into 2 tests:
"menu closes when disabled is toggled to true", and
"validation handler fires when disabled is toggled to true"

Reply:
Could wrap it in a `describe` too:
```
describe('when disabled is toggled to true', () => ... )
```